chain/chain/src/garbage_collection.rs

```diff
@@ -173,6 +181,14 @@ impl Chain {
                     break;
                 } else if prev_block_refcount == 1 {
                     debug_assert_eq!(blocks_current_height.len(), 1);
+                    if let Some(ref validator_account_id) = validator_account_id {
```

Comment:
Although it's true that only validators would ever need to do state sync (for now), we probably don't need to check that here; instead, the ShardTracker would provide the tracking requirements for both validators and non-validators, so we can use the same code path for both.